### PR TITLE
Damage System

### PR DESCRIPTION
**Pull request by aPunch**
Signed-off-by: aPunch theapunch@yahoo.com

This adds a better damage system. It takes into account the player's armor and whether the armor is enchanted. In the future it can be expanded upon to include potion effects.

I agree that this needs to be changed/fleshed out more, so provide any feedback that you have.

### CODE REVIEW

File: src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java [hunk 1]
```diff
@@ -246,7 +246,7 @@ private void updateHealthAndHunger() {
 			System.out.println("Hunger: " + hunger);
 			System.out.println("Health: " + health);
 			System.out.println("Exhaustion: " + exhaustion);
-			sendPacket(owner, new UpdateHealthMessage(health, hunger, foodSaturation));
+			// sendPacket(owner, new UpdateHealthMessage(health, hunger, foodSaturation));
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java:249]: Any reason you commented this out?

Comment (LunNova) [inline src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java:249]: I think there was an issue with it resulting in the hurt animation.

Comment (aPunch) [inline src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java:249]: I moved the UpdateHealthMessage to setHealth in VanillaPlayer. Also, there is an issue with regeneration where the hurt animation is played, but apparently that cannot be fixed in an easy manner until SpoutClient is ready. Am I correct on this, @NinjaZidane?

Comment (windy1) [inline src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java:249]: That can't be, it was working fine before. The rest of this PR looks fine so I'll go ahead and pull it and look into this later today. :)

**Pull request merged**